javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1392,24 +1392,37 @@ public T getNext() {
             final Iterator<T> that = this;
             return new AbstractIterator<T>() {
 
-                private T next = null;
+                private T next;
```

Comment:
I think my solution can be made faster - I will try... (but currently busy with the Gradle online course)